CorsixTH/Lua/humanoid_actions/seek_room.lua

```diff
@@ -246,115 +253,162 @@ local function action_seek_room_start(action, humanoid)
     return
   end
   -- Tries to find the room, if it is a diagnosis room, try to find any room in the diagnosis list.
-  local room = action_seek_room_find_room(action, humanoid)
+  if not humanoid.diagnosed or action.room_type == "research" then
+    local room = action_seek_room_find_room(action, humanoid)
+    -- if we have the room but not the staff we shouldn't seek out the room either
+    if room then
+      if humanoid.message then
+        TheApp.ui.bottom_panel:removeMessage(humanoid)
+      end
+      action_seek_room_goto_room(room, humanoid, action.diagnosis_room)
+      return
+    end
+  end
```

Comment:
Restructuring the code a bit here as one of the 'differences' between corsix and th. Seek diagnosis or research rooms, and remove the message if it exists (this cleans up interrupted seeks)